test/util/utils.js

```diff
@@ -398,21 +413,21 @@ async function setupTestEnvironment(contracts, { bosonTokenAddress, forwarderAdd
     signers: signers.slice(3),
     contractInstances,
     protocolConfig,
-    diamondAddress: protocolDiamond.address,
+    diamondAddress: await protocolDiamond.getAddress(),
     extraReturnValues,
   };
 }
 
 async function getSnapshot() {
-  return await ethers.provider.send("evm_snapshot", []);
+  return await provider.send("evm_snapshot", []);
 }
 
 async function revertToSnapshot(snapshotId) {
-  return await ethers.provider.send("evm_revert", [snapshotId]);
+  return await provider.send("evm_revert", [snapshotId]);
 }
 
 function deriveTokenId(offerId, exchangeId) {
-  return ethers.BigNumber.from(offerId).shl(128).add(exchangeId);
+  return BigInt(offerId) * 2n ** 128n + BigInt(exchangeId);
```

Comment:
```suggestion
  return BigInt(offerId) << 128n + BigInt(exchangeId);
```
I guess you could stick with a shift operation. It doesn't really matter, though.

Reply:
changed